src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.DateTime.netfx.cs

```diff
@@ -42,6 +42,17 @@ private class TimeSync
         private static TimeSync timeSync = new TimeSync();
 
         // sync DateTime and Stopwatch ticks every 2 hours
-        private static Timer syncTimeUpdater = new Timer(s => { Sync(); }, null, 0, 7200000);
+        private static Timer syncTimeUpdater = InitalizeSyncTimer();
+
+        [System.Security.SecuritySafeCritical]
+        private static Timer InitalizeSyncTimer()
+        {
+            // Don't capture the current ExecutionContext and its AsyncLocals onto the timer causing them to live forever
+            ExecutionContext.SuppressFlow();
```

Comment:
This is before-field-init static constructor that can be called pretty much any time. I am wondering whether this needs to be hardened against the case of flow suppressed already. Something like:

```
bool restoreFlow;
try
{
    if (!ExecutionContext.IsFlowSuppressed())
    {
        ExecutionContext.SuppressFlow();
        restoreFlow = true;
    }

    ...
}
finally
{
    if (restoreFlow)
        ExecutionContext.RestoreFlow();
}
```

Reply:
Good point; should probably do this with the others also https://github.com/dotnet/corefx/pull/26065, https://github.com/dotnet/corefx/pull/26066, https://github.com/dotnet/corefx/pull/26068, https://github.com/dotnet/corefx/pull/26073, https://github.com/dotnet/corefx/pull/26075, https://github.com/dotnet/corefx/pull/26077 as they are lazy first use triggered timers so context could also be already Supressed

Reply:
Done